### PR TITLE
Fixes #297. Single aws_resource_info per tag_selection metric block

### LINKED ISSUE #297: aws_resource_info while using aws_tag_select causes prometheus/expfmt parsing errors due to multiple same HELP/TYPE lines
I tried running the prometheus cloudwatch exporter with 2 configurations separately for EC2 CPU info

Scenario 1:
```
region: us-east-1
metrics:
- aws_namespace: AWS/EC2
  aws_metric_name: CPUUtilization
  aws_dimensions: [InstanceId]
  aws_tag_select:
    tag_selections:
      aws_service: ["opsworks"]
    resource_type_selection: "ec2:instance"
    resource_id_dimension: InstanceId
  aws_statistics: [Maximum]
```

Scenario 2:
```
region: us-east-1
metrics:
- aws_namespace: AWS/EC2
  aws_metric_name: CPUUtilization
  aws_dimensions: [InstanceId]
  aws_dimension_select:
    InstanceId: [i-X, i-Y, i-Z]
  aws_statistics: [Maximum]
```

The parsing of output in scenario 2 is fine but parsing the output of scenario 1 fails with error: `Invalid decode: text format parsing error in line 12: second HELP line for metric name "aws_resource_info".`

The instances filtered by the aws_tag_select in scenario 1 are identical to the ones specified inline in scenario 2.

### PR DESCRIPTION
**Pull request by mikhailadvani**
Signed-off-by: Mikhail Advani <mikhail.advani@gmail.com>

### CODE REVIEW

File: src/main/java/io/prometheus/cloudwatch/CloudWatchCollector.java [hunk 1]
```diff
@@ -640,6 +640,7 @@ private void scrape(List<MetricFamilySamples> mfs) throws CloneNotSupportedExcep
         }
         
         // Add the "aws_resource_info" metric for existing tag mappings
+        List<MetricFamilySamples.Sample> samples = new ArrayList<MetricFamilySamples.Sample>();
```
Comment (brian-brazil) [inline src/main/java/io/prometheus/cloudwatch/CloudWatchCollector.java:643]: This needs to be at the top of the function, before the for loop. A name like infoSamples would also be clearer.

File: src/main/java/io/prometheus/cloudwatch/CloudWatchCollector.java [hunk 2]
```diff
@@ -658,13 +659,13 @@ private void scrape(List<MetricFamilySamples> mfs) throws CloneNotSupportedExcep
...
-            List<MetricFamilySamples.Sample> samples = new ArrayList<MetricFamilySamples.Sample>();
+
             samples.add(new MetricFamilySamples.Sample("aws_resource_info", labelNames, labelValues, 1));
-            mfs.add(new MetricFamilySamples("aws_resource_info", Type.GAUGE, "AWS information available for resource", samples));
-            
+
             publishedResourceInfo.add(resourceTagMapping.getResourceARN());
           }
         }
+        mfs.add(new MetricFamilySamples("aws_resource_info", Type.GAUGE, "AWS information available for resource", samples));
```
Comment (brian-brazil) [inline src/main/java/io/prometheus/cloudwatch/CloudWatchCollector.java:668]: Similarly, this goes after the outer for.

**Comment by mikhailadvani** [comment]
can we have this merged & released please?

**Comment by brian-brazil** [comment]
Thanks!

I hadn't seen that you'd pushed, should be in the next week or so.

**Pull request merged**